### ISSUE TITLE
EventException trying to right-click on first play point

### ISSUE THREAD

**Issue by SlimeDog** [MartenM/ServerTutorialPlus#1]
Spigot 1.13.1
STP 1.19.0

My tutorial was working. Obviously I changed something, but I cannot determine what. After a server reboot, if I start the tutorial, I am transported to the first play point. When I try to right-click on it:
```
[22:27:00 ERROR]: Could not pass event PlayerInteractEvent to ServerTutorialPlus v1.19.0
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:308) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at org.bukkit.craftbukkit.v1_13_R2.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:240) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at net.minecraft.server.v1_13_R2.PlayerInteractManager.a(PlayerInteractManager.java:452) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1242) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at net.minecraft.server.v1_13_R2.PacketPlayInUseItem.a(PacketPlayInUseItem.java:37) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at net.minecraft.server.v1_13_R2.PacketPlayInUseItem.a(PacketPlayInUseItem.java:1) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.a(SourceFile:10) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_181]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_181]
        at net.minecraft.server.v1_13_R2.SystemUtils.a(SourceFile:199) [spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:900) [spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:835) [spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:733) [spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
Caused by: java.lang.NullPointerException
        at nl.martenm.servertutorialplus.points.custom.ClickBlockPoint$1$1.run(ClickBlockPoint.java:77) ~[?:?]
        at nl.martenm.servertutorialplus.managers.clickactions.ClickManager.handleClickAction(ClickManager.java:43) ~[?:?]
        at nl.martenm.servertutorialplus.events.OnPlayerInteractEvent.onPlayerInteractEvent(OnPlayerInteractEvent.java:30) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_181]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_181]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_181]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:304) ~[spigot-1.13.1.jar-2018-10-04-0610:git-Spigot-f6a273b-02518f9]
        ... 17 more
```

This EventException needs to be caught, in order to provide a more informative error message to the user. What is the underlying issue? Where is the click event failing? At what location?

**Comment by MartenM** [comment]
ClickBlockPoint.java:77 refernces to:
`animationTask.cancel();`

I checked the code for the class and it seems that if you disable the particle effects, no task is being created, thus giving a NPE when clicked.

**Comment by SlimeDog** [comment]
Well, that would explain it. Please add the obvious items to the testing protocol.

**Comment by SlimeDog** [comment]
Confirmed fixed in STP 1.19.2. Thanks.
